### PR TITLE
Add `expect` calls when async is involved

### PR DESCRIPTION
**Pull request by seanpdoyle**
`expect` calls within callbacks ensure that all expectations are invoked
and passing.

### CODE REVIEW

File: packages/ember-data/tests/unit/model/errors_test.js [hunk 1]
```diff
@@ -30,7 +30,6 @@ function unexpectedSend(eventName) {
 }
 
 test("add error", function() {
-  expect(6);
```
Comment (igorT) [inline packages/ember-data/tests/unit/model/errors_test.js]: Is there a downside to having the expect here?

**Comment by igorT** [comment]
Awesome work!

**Comment by igorT** [comment]
I think we should keep the expects though, so if the test gets refactored they are kept around

File: packages/ember-data/tests/unit/model/errors_test.js (hunk 1 above)
Comment (seanpdoyle) [inline packages/ember-data/tests/unit/model/errors_test.js]: Synchronous expects make the test

* noisy - if an expectation fails, the subsequent failures won't be run anyway
* brittle - adding/removing an additional expectation requires both adding/removing the expectation, as well as incrementing/decrementing `expect(n)`

Comment (igorT) [inline packages/ember-data/tests/unit/model/errors_test.js]: That is true, but I am more worried about people using tests as an example and then not adding expects, and then believing their tests pass by mistake. Or someone refactoring the test to be async. Not sure it's a reasonable fear though

**Comment by seanpdoyle** [comment]
@igorT fixed :+1:

**Comment by igorT** [comment]
Thanks!

**Pull request merged**